server/src/main/java/com/cloud/network/lb/LoadBalancingRulesManagerImpl.java

```diff
@@ -2231,11 +2234,21 @@ public Pair<List<? extends UserVm>, List<String>> listLoadBalancerInstances(List
                 continue;
             }
 
+            String userVmAsString = userVm.toString();
+
             boolean isApplied = appliedInstanceIdList.contains(userVm.getId());
-            if ((isApplied && applied) || (!isApplied && !applied)) {
-                loadBalancerInstances.add(userVm);
-                serviceStates.add(vmServiceState.get(userVm.getId()));
+            String isAppliedMsg = isApplied ? "is applied" : "is not applied";
+            s_logger.debug(String.format("The user VM [%s] %s to a rule of the load balancer [%s].", userVmAsString, isAppliedMsg, loadBalancerAsString));
+
+            if (isApplied != applied) {
+                s_logger.debug(String.format("Skipping add service state from the user VM [%s] to the service state list. This happens because the VM %s to the load "
+                        + "balancer rule and the [%s] parameter was passed as [%s].", userVmAsString, isAppliedMsg, ApiConstants.APPLIED, applied));
```

Comment:
```suggestion
                if (s_logger.isDebugEnabled()) {
                    s_logger.debug(String.format("Skipping add service state from the user VM [%s] to the service state list. This happens because the VM %s to the load "
                        + "balancer rule and the [%s] parameter was passed as [%s].", userVmAsString, isAppliedMsg, ApiConstants.APPLIED, applied));
               }
```